### PR TITLE
feat: access control stubs

### PR DESCRIPTION
**Pull request by dbcfd**
add api stubs for supporting access control behind a feature flag

### CODE REVIEW

**Comment by nathanielc** [comment]
@dbcfd Do you have any design docs explaining these API endpoints?

File: api/ceramic.yaml [hunk 1]
```diff
@@ -495,3 +575,17 @@ components:
         streamId:
           type: string
           description: Multibase encoded stream ID.
+    RefreshTokenRequest:
+      title: A request for a refresh token
+      type: object
+      properties:
```
Comment (dav1do) [inline api/ceramic.yaml:581]: should the `requestId` property be required on this and AccessTokenRequest?

File: api/ceramic.yaml [hunk 2]
```diff
@@ -15,6 +15,60 @@ servers:
...
+    post:
+      summary: Create a refresh token
+      parameters:
+        - name: Authorization
+          in: header
+          description: Bearer token
+          required: true
+          schema:
+            type: string
+        - name: Did
```
Comment (dav1do) [inline api/ceramic.yaml:28]: should this header be `X-Did` or similar throught? (or X-Did... something that's not so easily conflated with the word `did` 🤷). have a custom header called `Did` seems unusual

File: api/src/server.rs [hunk 3]
```diff
@@ -312,6 +320,27 @@ where
             model,
             insert_task,
             marker: PhantomData,
+            feature_flags: HashSet::default(),
+        }
+    }
+
+    pub fn enable_feature(&mut self, feature: &str) {
```
Comment (dav1do) [inline api/src/server.rs:327]: longer term I image feature flags living inside the service crate or similar so they can be shared by recon and other parts of the system but it's implementation details and fine for now

**Review by dav1do — COMMENTED** [review]
I don't know if I see the point in adding routes that just panic 😂... but if we have a follow up to add the details I don't really have concerns with POC refresh/access token routes and accepting optional headers to validate.. minor suggestions to use the custom header `X-` prefix and maybe making properties required on the output

File: api/ceramic.yaml (hunk 1 above)
Comment (dbcfd) [reply api/ceramic.yaml:581]: At this point, I don't have enough information to really put anything in this structure, so would like to leave it empty, but openapi doesn't allow that. So I added requestId as an optional parameter, for tracing, even though that might be better as a header.

I feel like I might need list of resources and actions for a refresh token, but wasn't confident enough to add them in this PR.

File: api/ceramic.yaml (hunk 2 above)
Comment (dbcfd) [reply api/ceramic.yaml:28]: X-Did is probably better, will change.

File: api/src/server.rs (hunk 3 above)
Comment (dbcfd) [reply api/src/server.rs:327]: Yeah, there's better crates for feature flags, we may want to do a quick survey and find one that meets our needs before this or other features are done.

File: one/src/lib.rs [hunk 4]
```diff
@@ -149,6 +149,14 @@ struct DaemonOpts {
...
     )]
     idle_conns_timeout_ms: u64,
+
+    /// Enable authentication feature
+    #[arg(
+        long,
+        default_value_t = false,
+        env = "CERAMIC_ONE_ENABLE_AUTHENTICATION"
+    )]
+    enable_authentication: bool,
```
Comment (dav1do) [inline one/src/lib.rs:159]: didn't think about it first pass, but wonder if we'd want to make features flags and enum we could pass multiple times or as a comma separated string... A bool seems find for now, though we should probably mark these things unstable before GA.

**Review by dav1do — APPROVED** [review]
Okay from my perspective, the only concern I guess is removing flags from the CLI but as long as we make more progress on this before GA it's not an issue, plus I doubt anyone will use the flags we don't suggest as there are so many.

**Review by nathanielc — COMMENTED** [review]
Can you help me understand why we want these stubs in place before implementing them? Are you looking for feedback on the design of the API?

I agree with @dav1do we should not change the CLI flags yet until we know what/how they will be used.

We are also working to publish the generated clients for this API. As such having these partially defined endpoints means confusion for end users about the failing endpoints and future breaking changes when we do change them to be complete.

**Pull request closed without merging**